geoalchemy2/__init__.py

```diff
@@ -103,19 +102,32 @@ def dispatch(event, table, bind):
                         stmt = stmt.execution_options(autocommit=True)
                         bind.execute(stmt)
                     elif bind.dialect.name == 'postgresql':
+
+                        index_name = "idx_{}_{}".format(table.name, c.name)
+
+                        if c.type.use_N_D_index:
+                            gis_column = '{} gist_geometry_ops_nd'.format(c.name)
+                        else:
+                            gis_column = c.name
+
                         if table.schema:
-                            q = text('CREATE INDEX "idx_%s_%s" ON "%s"."%s" '
-                                     'USING GIST ("%s")' %
-                                     (table.name, c.name, table.schema,
-                                      table.name, c.name))
+                            table_name = "{}.{}".format(table.schema, table.name)
                         else:
-                            q = text('CREATE INDEX "idx_%s_%s" ON "%s" '
-                                     'USING GIST ("%s")' %
-                                     (table.name, c.name, table.name, c.name))
+                            table_name = table.name
+
+                        sql = "CREATE INDEX {} ON {} USING GIST ({})".format(index_name, table_name,
+                                                                             gis_column)
+
+                        q = text(sql)
+
                         bind.execute(q)
                     else:
                         raise ArgumentError('dialect {} is not supported'.format(bind.dialect.name))
 
+                if isinstance(c.type, (Geometry, Geography)) and c.type.spatial_index is False and \
+                        c.type.use_N_D_index is True:
```

Comment:
```suggestion
                if c.type.spatial_index is False and c.type.use_N_D_index is True:
```
I think that `use_N_D_index` should never be `True` when `spatial_index` is `False`, regardless of the type of `c.type`. WDYT?

Reply:
Correct, this just checks that state.
A potential better way of handling this could be done under the hood and set the 'use_N_D_index' flag automatically to False rather than raising an error.

Reply:
Something like this?
```
if c.type.spatial_index is False:
    c.type.use_N_D_index = False
```